File: Demonstrators/FakeClusterFinder/bin/runClusterFinding.cc
#include "DataFormat/storage_manager.h"
#include "FakeClusterFinder/ClusterFinderHelper.h"

int main(){

  larlite::storage_manager storage;

  storage.set_verbosity(larlite::msg::kDEBUG);

  storage.set_io_mode(larlite::storage_manager::kBOTH);

  storage.add_in_filename("trial.root");
  storage.set_out_filename("fakeclusters.root");

  storage.open();

  while (storage.next_event()) {

    storage.set_id(storage.run_id(), storage.subrun_id(), storage.event_id());

    demonstrators::ClusterFinderHelper helper;

    helper.findClusters(storage, "test");
  }

  storage.close();

  return 0;
}

File: Demonstrators/FakeClusterFinder/mac/runClusterFinding.py
import sys
from ROOT import larlite
import ROOT

man=larlite.storage_manager()
man.set_io_mode(man.kBOTH)

# you can generate the input file using
# core/DataFormat/bin/simple_write.cc
man.add_in_filename("trial.root")

man.set_out_filename("fakeclusters.root")
man.set_verbosity(larlite.msg.kDEBUG)
man.open()

while man.next_event() :

    man.set_id(man.run_id(), man.subrun_id(), man.event_id())

    helper = ROOT.demonstrators.ClusterFinderHelper()

    helper.findClusters(man, "test")

man.close()
